src/tests/Microsoft.PowerFx.Json.Tests/JSONTests.cs

```diff
@@ -36,5 +37,52 @@ public void Json_IncludeBinaryData_NoSideEffects()
             Assert.False(result.IsSuccess);
             Assert.Equal("The JSON function cannot serialize binary data in non-behavioral expression.", result.Errors.First().Message);
         }
+
+        public class LazyRecordType : RecordType
```

Comment:
Please add a test where there's a recursive type & value, validate that it errors after a certain depth